velox/dwio/parquet/reader/CompressionQpl.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <qpl/qpl.h>
+#include <iostream>
+#include <memory>
+
+bool Initjobs(qpl_path_t execute_path);
+class Qplcodec {
+ public:
+  Qplcodec(qpl_path_t execute_path, qpl_compression_levels compression_level)
+      : execute_path_(execute_path),
+        compression_level_(compression_level),
+        job_(NULL),
+        idx_(0){};
+  ~Qplcodec() {
+    // Freejob();
```

Comment:
@Yuhta Yeah, there is a memory leak problem. When velox calls QPL:decompress function first time, it need to malloc some job pool. So, the later function call will not malloc job pool, but when velox exits, we need to free these job pool. But at present, I don`t know where to free these job pool. It will be helpful if you can give some suggestions. I think, Velox also need to free some global memory space, the job pool free can be put there.

Reply:
How large is the malloc and initialization cost?  For other decompression we just release the memory on each invocation and usually this is not a bottleneck for performance.  Can you do a simple benchmark and see if it is feasible to do the same?

Reply:
Thanks, the job pool size is about 120KB for hardware path or 700KB for software path. The initialization cost is high if we release the memory on each invocation. This finding has been proven on other workloads, such as oceanbase. If we don`t release these memory, [**job pool count * job pool size**] (KB) memory may lead to memory leak.

Reply:
Thanks for your comment, it`s a good idea, I will have a try.